### PR TITLE
Replace the existing path template parser with a recursive descent parser.

### LINKED ISSUE #11: Refactor path template to match other languages
Particularly, use go's yacc to generate the parser.

### PR DESCRIPTION
**Pull request by jba**
Also added a small test.

Resolves #11

### CODE REVIEW

**Comment by saicheems** [comment]
This is awesome! Thanks for writing it

**Comment by saicheems** [comment]
Nit: update the commit message to follow this format: http://chris.beams.io/posts/git-commit/ (imperative mood, < 50 char subject, wrap body at 72 char, etc.)

File: path_template_parser.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		p.error("expected '}'")
+	}
+	return segs
+}
+
+// A literal is any sequence of characters other than a few special ones.
+// The list of stop characters is not quite the same as in the template RFC,
+// but is identical to the list used in the custom verb regexp.
+func (p *pathTemplateParser) literal() string {
+	lit := p.consumeUntil(":/*}{=")
```
Comment (saicheems) [inline path_template_parser.go]: We have decided to allow `:`
For example, in logging, a project id of `example.com:hello-world` is valid.

Comment (saicheems) [inline path_template_parser.go]: In the process, we also decided to remove custom verbs entirely. There's an ambiguity introduced when parsing resource names that include a colon.

Comment (jba) [inline path_template_parser.go]: So should I remove all the tests involving custom verbs in path_template_test.go?

Comment (saicheems) [inline path_template_parser.go]: Yes

**Comment by jba** [comment]
@saicheems No rush on this, but I don't know if you were notified about my new pushes.

**Comment by saicheems** [comment]
Can also remove the `customVerb` field in the `PathTemplate` struct and in the `Match` function

Otherwise LGTM.
FYI github only sends notifications if a comment is made.

**Comment by jba** [comment]
Done.

What is the right way to merge? I want to squash and write a new commit message.

**Comment by saicheems** [comment]
Two ways:

You can squash locally and force push to your branch, or you can use github's squash feature - when you press merge pull request it'll ask you to enter a commit title and message. I've set this repo to only accept squashed commits.

**Pull request merged**